### PR TITLE
Add select shows form and search input 

### PR DESCRIPTION
**Pull request by Caruki**
Add search icons and search input form
Create functionality for search and add shows from search form to watch list
Add useThrottle hook and implement it into the search form 

Add search input to side navigation that opens the detail view on select.

See the live preview here:
https://show-cased-p-search-inp-ctdlzn.herokuapp.com/login
Example Account:
email: hallo@hallo.com
pw: hallihallo

### CODE REVIEW

File: client/src/assets/SearchIcon.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+          x2="0.5"
+          y2="1"
+          gradientUnits="objectBoundingBox"
+        >
+          <stop className="offset0" offset="0" />
+          <stop offset="1" stopColor="#aeb2f5" />
+        </linearGradient>
+      </defs>
+      <path
+        id="search"
```
Comment (lmachens) [inline client/src/assets/SearchIcon.js:36]: do you need this `id`?

File: lib/routes/users.js [hunk 2]
```diff
@@ -64,6 +65,28 @@ router.post('/:id/towatch', async (request, response) => {
...
+      userId,
+      {
+        $addToSet: { towatch: { $each: shows } },
+      },
+      { new: true }
+    );
+    if (updatedUser) {
+      response.send(updatedUser.towatch);
+    } else {
+      response.status(400).send('No user found!');
```
Comment (lmachens) [inline lib/routes/users.js:83]: Not found is usually `404`

File: lib/routes/users.js [hunk 3]
```diff
@@ -64,6 +65,28 @@ router.post('/:id/towatch', async (request, response) => {
   }
 });
 
+router.post('/:id/towatch/multi', async (request, response) => {
```
Comment (lmachens) [inline lib/routes/users.js:68]: `/multi` is not RESTful.
Instead, you would use `/api/users/:userId/towatch` as route and check if you added multiple showIds or a single showed. 
See https://stackoverflow.com/questions/15370387/restful-api-endpoint-for-adding-removing-array-elements

Comment (Caruki) [reply lib/routes/users.js:68]: good point! thanks for the heads up. I changed the endpoint and refactored a bit

**Pull request merged**